Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -90,165 +83,1086 @@ function Sync-AzSOfflineMarketplaceItem {
+        $productInfo['billingPartNumber'] = $productDetails.properties.billingPartNumber
+        $productInfo['vmExtensionType'] = $productDetails.properties.vmExtensionType
+        $productInfo['legalTerms'] = $productDetails.properties.description
+        $productInfo['payloadLength'] = $productDetails.properties.payloadLength
+        $productInfo['galleryItemIdentity'] = $productDetails.properties.galleryItemIdentity
+        $productInfo['productKind'] = $productDetails.properties.productKind
+        $productInfo['productProperties'] = $productDetails.properties.productProperties
+        $productDetailsProperties = @{}
+        $productDetailsProperties['version'] = $downloadDetails.properties.version
+        if ($downloadDetails.productKind -in ('solution', 'resourceProvider')){
+            $productInfo['dependentProducts'] = $downloadDetails.properties.dependentProducts
+            $containerIds = @()
+            $containerIds += $downloadDetails.properties.fileContainers.Id
+            $productInfo['fileContainers'] = $containerIds
+            $productDetailsProperties['dependentProducts'] = $downloadDetails.properties.dependentProducts
+            $productDetailsProperties['fileContainers'] = $downloadDetails.properties.fileContainers
+        } elseif ($downloadDetails.productKind -eq 'virtualMachine') {
+            $productDetailsProperties['OsDiskImage'] = $downloadDetails.properties.OsDiskImage
+            $productDetailsProperties['DataDiskImages'] = $downloadDetails.properties.DataDiskImages
+        } else {
+            $productDetailsProperties['vmOsType'] = $downloadDetails.properties.vmOsType
+            $productDetailsProperties['sourceBlob'] = $downloadDetails.properties.sourceBlob
+            $productDetailsProperties['computeRole'] = $downloadDetails.properties.computeRole
+            $productDetailsProperties['vmScaleSetEnabled'] = $downloadDetails.properties.vmScaleSetEnabled
+            $productDetailsProperties['supportMultipleExtensions'] = $downloadDetails.properties.supportMultipleExtensions
+            $productDetailsProperties['isSystemExtension'] = $downloadDetails.properties.isSystemExtension
+        }
+
+        $productInfo['links'] = $productDetails.properties.links
+        $productInfo['iconUris'] = $productDetails.properties.iconUris
+        $productInfo['galleryPackageBlobSasUri'] = $downloadDetails.galleryPackageBlobSasUri
+
+        $productDetails.properties|select publisherIdentifier,offer,sku,productKind,vmExtensionType  |out-file "$productFolder\$azpkgName.txt" -Append
+        $productDetails.properties.productProperties|select version| out-file "$productFolder\$azpkgName.txt" -Append
+
+        # select premium download
+        Write-Host $("-"*20)
+        $downloadConfirmation = Read-Host "Downloading package files. Would you like to use Premium download? This requires Azure Storage Tools to be installed. (Y/N)?"
```

Comment:
Generally there should only be a single prompt at the beginning, you can look-up the ShouldProcess pattern for using the built-in confirmation capabilities in PowerShell which can be appropraate for such things and also supports "Yes to all"

Reply:
Will work on this later.